UniversalMediaServer/UniversalMediaServer
# v14.1

# Description of code changes

Implements latest Jetty (v12.0.8).
Implements latest JUPnP (v3.0.1).
Removed the JDK HTTP servlet server (personal implementation).
Stop using the deprecated tool 'Create React App' (no longer being maintained)
Implements Vite as replacement.

The webgui / webplayer are now served by Jetty (servlet).

# Goal of the version

Remove multiple engine maintenance.
Remove Netty without stress.  #4058

# To test
We should focus on the webgui / webplayer tests first, as it is now served by Jetty servlets.
It will be easy to test as it is not changing settings / database or so that will break if we revert to v14 build.

Netty is kept as the default engine, letting us time to fix Jetty implementation (by changing the default engine in the GUI settings).
Once Jetty implementation is fully tested, we can remove all Netty related things (engine, deps, and so on).

# Checklist
- [x] Any relevant issues are [referenced](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#referencing-issues-and-pull-requests)
- [x] Any relevant [Knowledge Base](https://github.com/UniversalMediaServer/knowledge-base) articles are written/modified
- [x] Any relevant tests have been written/modified
